storage/block.go

```diff
@@ -43,7 +44,8 @@ type dbBlock struct {
 	opts     m3db.DatabaseOptions
 	start    time.Time
 	encoder  m3db.Encoder
-	stream   m3db.SegmentReader
+	segment  *m3db.Segment
```

Comment:
Any reason to make this a pointer?  I'd be more in favor of having it as just a struct, considering its just two pointers inside the struct anyway.  Would prefer to relieve pressure on the garbage collector where possible with less objects to track.